Docs: Added a How-to guide to use API key (#1667)
hirosystems/stacks-blockchain-api: content/how-to-guides/how-to-use-api-key.md

---
Title: How to use API key
---

# Steps to use API Key

This guide helps you with the steps to use the API key to interact with the Stacks Blockchain API.

> 1. [Send a request](https://survey.hiro.so/hiroapi) to the Hiro team to get an API key. You will get the response in the form of the following:

Author: @sabbyanandan Could you help with the correct link to the request form?

> 1. [Send a request](https://survey.hiro.so/hiroapi) to the Hiro team to get an API key. You will get the response in the form of the following:

Reviewer: Hi. This form is to gather interest. I am confused why we are adding it here as part of "[How-to guide to use API key]" guide. We don't need it here.

This guide should be about how a dev will use an API to make the HTTP requests, which is captured in the CURL command below. 

Also show another example of how to do it in a Typescript function for additional guidance.

```
NAME: x-hiro-api-key
VALUE: Nabcs1234efg56789aaaaaaaaqqqqqqqeeeeee12334345
```

There are multiple ways to interact with the endpoint. However, this document walks you through two ways to make an [API call](https://docs.hiro.so/api#tag/Microblocks/operation/get_microblock_list). You can choose between any of the following methods.

- Using Curl
- Using Postman

## Using Curl

The following section walks you through using an API key with Curl.

`curl https://api.hiro.so -H 'x-hiro-api-key: <API_KEY>'`

Now, put the API key value that you received from your request, and the command now looks like this: 

> `curl https://api.hiro.so/extended/v1/microblock -H 'x-hiro-api-key: Nabcs1234efg56789aaaaaaaaqqqqqqqeeeeee12334345'`

Author: Done!

> `curl https://api.hiro.so/extended/v1/microblock -H 'x-hiro-api-key: Nabcs1234efg56789aaaaaaaaqqqqqqqeeeeee12334345'`

Reviewer 2: Perfect! Thanks!

---
Title: How to use API key
---

> API keys are alpha-numeric string values distributed to the developer customers to grant access to your API.

Reviewer: ```suggestion
API keys are alpha-numeric codes that identify and authenticate an application or developer.
```

# Steps to use API Key

> This guide helps you with the steps to use the API key to interact with the Stacks Blockchain API.

Reviewer: As discussed, let's generalize this document in its entirety, given that this guide applies to all 3 APIs we ship/support. 

We may have to find a way to reuse this guide for all 3 product sections, also.

:::note

_The API key is passed in the **header** of your API call._

:::

> There are multiple ways to interact with the endpoint. However, this document walks you through three ways to make an [API call](https://docs.hiro.so/api#tag/Microblocks/operation/get_microblock_list). You can choose between any of the following methods.

Reviewer: ```suggestion
There are multiple ways to interact with the API endpoints. However, this document walks you through three ways to make an [API call](https://docs.hiro.so/api#tag/Microblocks/operation/get_microblock_list). You can choose between any of the following methods.
```

- Curl
- Postman
- Typescript function

## Using Curl

Using Curl, you will pass the API key in an `x-hiro-api-key` header. Use the following command as an example to call the API endpoint `https://api.hiro.so/extended/v1/microblock`.

`curl https://api.hiro.so/extended/v1/microblock -H 'x-hiro-api-key: Nabcs1234efg56789aaaaaaaaqqqqqqqeeeeee12334345'`

## Using Postman

The following section walks you through using an API key with [Postman](https://www.postman.com/).

In the Postman request, add the [API endpoint](https://api.hiro.so/extended/v1/microblock) with the `Get` request. Then, add the API key and its value received from the Hiro team in the **Headers** section, as shown in the image below.

Select **Send** to get recent microblocks.

![API-Key](../images/api-key.jpeg)

> ## Using the Typescript function

Reviewer: ```suggestion
## Using the API key in Typescript
```

Reviewer: @rafaelcr: You may have some commentary on the TS function. Feel free to suggest-edit.

Reviewer 3: Code sample LGTM
